### ISSUE TITLE
Add a button to the admin UI to allow administrators to restart services without command line access

### ISSUE THREAD

**Issue by mike-r-mclaughlin** [sourcegraph/sourcegraph#40625]
**Requested by:** https://github.com/sourcegraph/accounts/issues/6716, https://github.com/sourcegraph/accounts/issues/554

#### Feature request description

When making changes to the the configuration, a server restart is [sometimes](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/conf/parse.go?L49:12) required.  The UI gives a warning like: `Server restart is required for the configuration to take effect.`.  

#### Is your feature request related to a problem? If so, please describe.

1. It's not clear what "server restart" means when deployed in K8s or Docker Compose
2. Large customers often don't have command line access to production environments

#### Additional context

[Slack discussion](https://sourcegraph.slack.com/archives/C022SPMNR0W/p1656598458060419)

**Comment by keegancsmith** [comment]
Looking at the list of config vars that require restart, I wonder if we could move to a world were most of them no longer require restart? Restart feels like a hack.

**Comment by scjohns** [comment]
@keegancsmith yeah that'd be ideal. I think there's multiple end states of varying degrees of work that are all acceptable, and I would imagine the work required to fix is aligned with how optimal the solution is:

1. No restart is required (ideal)
2. Add a button to the UI to be able to restart when necessary and a checkbox somewhere to say "auto restart frontend for config update"
3. Update the message in the UI to look at the deployment type and offer a command/steps to restart (easiest, least overall helpful)

I also recall hearing at one point, I think there are some configs that trigger the message that don't actually require a restart to take effect, which isn't a great UX.
